### ISSUE TITLE
Add an archlinux pkg build

### ISSUE THREAD

**Issue by offbeat-stuff** [fn2006/PollyMC#24]
### Role

I run archlinux and i would be happy if you included this in the build

### Suggestion

I want you to add a archlinux package to the releases

### Benefit

I would help archlinux users to use your software

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

this is the pkgbuild I downloaded from aur/polymc-bin and changed the urls and skipped the checksums to install pollymc you can use this to provide the pkg
```
# Maintainer: Lenny McLennington <lenny@sneed.church>
# Maintainer: Sefa Eyeoglu <contact@scrumplex.net>

pkgname=pollymc-bin
pkgver=1.4.2
pkgrel=1
pkgdesc="Minecraft launcher with ability to manage multiple instances."
arch=('x86_64')
url="https://github.com/fn2006/PollyMC"
license=('GPL3')
depends=('java-runtime' 'libgl' 'qt6-base' 'qt6-5compat' 'qt6-svg' 'qt6-imageformats' 'zlib' 'hicolor-icon-theme')
provides=('polymc')
conflicts=('polymc')
optdepends=('java-runtime=8: support for Minecraft versions < 1.17'
            'java-runtime=17: support for Minecraft versions >= 1.17')
source=("https://github.com/fn2006/PollyMC/releases/download/${pkgver}/PollyMC-Linux-Qt6-${pkgver}.tar.gz")
noextract=("PollyMC-Linux-Qt6-${pkgver}.tar.gz")
#sha256sums=('6958a77c5b99a8c1ec1b1ef50982c8668f076c29c450b796b5a0aeb33ac8a95c')

package() {
    install -d "$pkgdir/usr"
    tar -C "$pkgdir/usr" -xvf PollyMC-Linux-Qt6-${pkgver}.tar.gz
    chown -R root:root "$pkgdir/usr"  # fils in tarball are not owned by root
}
```

**Comment by fn2006** [comment]
Yeah that sounds like a good idea, PollyMC is already on the AUR but I might also include an arch package in the releases for those who don't like compiling.

**Comment by AndreAugustoAAQ** [comment]
It's a great addition to put it in the AUR
It would be interesting to use actions to automate this process
Note that this provided pkgbuild needs several modifications
I believe that PollyMC does not conflict with PolyMC
And we also have to pay attention to the [arch packaging guidelines](https://wiki.archlinux.org/title/Arch_package_guidelines)

**Comment by Samisafool** [comment]
> PollyMC is already on the AUR

```
$ paru -Ss pollymc
aur/pollymc-qt5-git 1.1.1.r1135.g49849b75-1 [+2 ~1.23]
    Fork of PolyMC that adds Ely.by support and allows you to use offline mode without an account.
```

Only the git version though.

```
# Maintainer: Lenny McLennington <lenny@sneed.church>
# Maintainer: Sefa Eyeoglu <contact@scrumplex.net>

pkgname=pollymc-bin
pkgver=1.4.2
pkgrel=1
pkgdesc="Minecraft launcher with ability to manage multiple instances."
arch=('x86_64')
url="https://github.com/fn2006/PollyMC"
license=('GPL3')
depends=('java-runtime' 'libgl' 'qt6-base' 'qt6-5compat' 'qt6-svg' 'qt6-imageformats' 'zlib' 'hicolor-icon-theme')
optdepends=('java-runtime=8: support for Minecraft versions < 1.17'
            'java-runtime=17: support for Minecraft versions >= 1.17')
source=("https://github.com/fn2006/PollyMC/releases/download/${pkgver}/PollyMC-Linux-Qt6-${pkgver}.tar.gz")
noextract=("PollyMC-Linux-Qt6-${pkgver}.tar.gz")
#sha256sums=('6958a77c5b99a8c1ec1b1ef50982c8668f076c29c450b796b5a0aeb33ac8a95c')

package() {
    install -d "$pkgdir/usr"
    tar -C "$pkgdir/usr" -xvf PollyMC-Linux-Qt6-${pkgver}.tar.gz
    chown -R root:root "$pkgdir/usr"  # files in tarball are not owned by root
}
```

**Comment by fn2006** [comment]
Added regular and -bin versions to the AUR.

**Comment by Samisafool** [comment]
Does it conflict with polymc? I don't think it does

**Comment by fn2006** [comment]
I didn't think they conflicted either but then I tried installing them alongside each other and it failed because they both install jars to `/usr/share/jars/`.
As for the java-runtime optdepend, it's meant to be =17, Java 17 is an LTS release so Minecraft will likely be using it for the forseeable future, even though Java 18 might work with Minecraft it's not recommended and future Java versions might cause issues.

**Comment by Snogard** [comment]
In august there was a package named "polymc-cracked" and was removed from the AUR citing Microsoft EULA violation.
links [here](https://lists.archlinux.org/archives/list/aur-requests@lists.archlinux.org/message/6QWJJ3USDUA6O2CVJ5R7IFLQLYYKOGFW/) and [here](https://lists.archlinux.org/archives/list/aur-requests@lists.archlinux.org/message/SJUTSZE4ND2SJDSMJ7H2ORMRIVP2TAPL/)

**Comment by yum13241** [comment]
Maybe a custom repo could work.

**Comment by RevaldoVeryCool** [comment]
why not make a flatpak version? as there are tl:legacy on flathub, might be possible to add it to flathub without issue

**Comment by yum13241** [comment]
Because flatpaks have many issues LIKE THEIR F**KING STUPID NAMES!!!111!!!!eleven!

**Comment by fn2006** [comment]
> why not make a flatpak version? as there are tl:legacy on flathub, might be possible to add it to flathub without issue

I'm planning on including a flatpak build in 7.0, I can't be bothered to put it on flathub though.

**Comment by RevaldoVeryCool** [comment]
> Because flatpaks have many issues LIKE THEIR F**KING STUPID NAMES!!!111!!!!eleven!

i didn't have any issue with flatpak, prob just me

> > why not make a flatpak version? as there are tl:legacy on flathub, might be possible to add it to flathub without issue
> 
> I'm planning on including a flatpak build in 7.0, I can't be bothered to put it on flathub though.

oh alright. im not bothered by the build is not in flathub too ¯\_(ツ)_/¯

**Comment by yum13241** [comment]
> i didn't have any issue with flatpak, prob just me

With native packages, their names are simple. `pollymc` as an example.

With flatpaks, it'd be something like `com.fn2006.pollymc`, which also requires `flatpak run` to run it from the command line. Not to mention the sandboxing which makes it impossible to share setups between different OSes.

**Comment by luni3359** [comment]
Just go inside ~/.var/app and access the files you need to from the pollymc directory.

**Comment by RevaldoVeryCool** [comment]
> > i didn't have any issue with flatpak, prob just me
> 
> With native packages, their names are simple. `pollymc` as an example.
> 
> With flatpaks, it'd be something like `com.fn2006.pollymc`, which also requires `flatpak run` to run it from the command line. Not to mention the sandboxing which makes it impossible to share setups between different OSes.

oh alright, im noob at linux xD

**Comment by yum13241** [comment]
> Just go inside ~/.var/app and access the files you need to from the pollymc directory.

`~/.var/app` is one of the most annoying directories ever, since it means my configs are fragmented. I have a shared NTFS drive between Linux & Windows, and I set PollyMC to use the same instances n the shared drive. Sandboxing makes that impossible. AppImage ftw.

**Comment by offbeat-stuff** [comment]
yeah but pollymc can sandbox the running minecraft process further you can preceed minecraft command with firejail or smth in pollymc settings

**Comment by Vilianexe** [comment]
> > Just go inside ~/.var/app and access the files you need to from the pollymc directory.
> 
> `~/.var/app` is one of the most annoying directories ever, since it means my configs are fragmented. I have a shared NTFS drive between Linux & Windows, and I set PollyMC to use the same instances n the shared drive. Sandboxing makes that impossible. AppImage ftw.
> 
> 1 positive would be that PollyMC works on the Steam Deck, at least in theory.

you can just disable the sandbox for the path of your mounted driver, you are just lying about it being impossible

**Comment by yum13241** [comment]
My other points still stand. Dumb terminal name, dumb theming, and if you use `fish` as your shell then Flatpaks won't appear in your app menu unless you restart. I blame `fish` for this one though. It ain't `POSIX` complaint, goddamnit!
